src/Plotting/plot_recipes.jl

```diff
@@ -161,6 +162,56 @@ julia> plot(B, 1e-2)  # faster but less accurate than the previous call
         seriesalpha --> DEFAULT_ALPHA
         seriescolor --> DEFAULT_COLOR
 
+        # extract limit and extrema of plotted shapes if existings
+        lims = Dict()
+        extr = Dict()
+        p = plotattributes[:plot_object]
+        if length(p) > 0
+            subplot = p[1]
+            for symbol in [:x, :y]
+                lims[symbol] = subplot[Symbol(symbol,:axis)][:lims]
+                extr[symbol] = subplot[Symbol(symbol,:axis)][:extrema]
+            end
+        end
+
+        if !isbounded(X)
+            # if the limit is :auto, set the limits to the current extrema
+            for symbol in [:x, :y]
+                if lims[symbol] == :auto
+                    emin = extr[symbol].emin
+                    emax = extr[symbol].emax
+                    # if the extrema is (-Inf,Inf), i.e. an empty plot,
+                    # set it to (0.,1.)
+                    lmin = isinf(emin) ? 0.0 : emin
+                    lmax = isinf(emax) ? 1.0 : emax
+                    lims[symbol] = (lmin, lmax)
+                    # set the limit of the plot
+                    subplot[Symbol(symbol,:axis)][:lims] = lims[symbol]
+                end
+            end
+            # otherwise keep the old limits
+
+            # bound unbounded set
+            low_lim = [lims[:x][1] - DEFAULT_PLOT_LIMIT, lims[:y][1] - DEFAULT_PLOT_LIMIT]
+            high_lim = [lims[:x][2] + DEFAULT_PLOT_LIMIT, lims[:y][2] + DEFAULT_PLOT_LIMIT]
+            X = intersection(X, Hyperrectangle(low=low_lim, high=high_lim))
+
+        # if there is already a plotted shape and the limits are fixed,
+        # automatically adjust the  axis limits (e.g. after plotting a unbounded set)
+        elseif  length(p) > 0 && lims[:x] != :auto && lims[:y] != :auto
+            # update limits such that the new set is contained in the plot windows
+            box = box_approximation(X)
+            min_box = low(box)
+            max_box = high(box)
+            for (idx,symbol) in enumerate([:x, :y])
+                # if the new set is not contained in limits, extend the limit
+                ϵ = 0.05
+                min_lim = min(lims[symbol][1], min_box[idx]*(1 - ϵ))
+                max_lim = max(lims[symbol][2], max_box[idx]*(1 + ϵ))
```

Comment:
1. The multiplication only makes sense for sets in the positive quadrant.
2. A relative scaling means that the same sets shifted by a large scalar will result in different plots.
3. For checking whether the current bounds should be extended, this additional offset should not be considered.

In the interest of simplicity, I would just ignore this beautification.

Reply:
> 1. The multiplication only makes sense for sets in the positive quadrant.

Fixed it by 
```
            ϵ = 0.05
            min_offset = abs(min_box[idx])*ϵ
            max_offset = abs(max_box[idx])*ϵ
            min_lim = min(lims[symbol][1], min_box[idx] - min_offset)
            max_lim = max(lims[symbol][2], max_box[idx] + max_offset)
```

> 2. A relative scaling means that the same sets shifted by a large scalar will result in different plots.
> 3. For checking whether the current bounds should be extended, this additional offset should not be considered.

 I disagree with these points. I am under the opinion that the plot offset should be dependent on the plot size and not the set size.  I really liked the resulting extension of the plot limits by this implementation. However, I did not test the plotting behaviour entierly, so if there are reasons why my implementation doesn't make sense I gladly change it.

Reply:
Take the values `min_box[idx] == 0` and `max_box[idx] == 1000`. Then `min_offset == 0` and `max_offset == 50`. I doubt that this is what you want.

Reply:
Okay, I see you are totally right, I didn't think this through. But I think we are on the right track. I will just keep trying for the moment and keep your examples as a test case.

Reply:
I think you want to have the *width* (max - min) and scale that.

Reply:
I am not sure anymore, what I really want 😅
It's close to what you proposed. But I think I will use the width of the entire plotting windows and not only the set.

Reply:
That's fine :+1: